### PR TITLE
[docker-teamd]: Manage teamd and teamsyncd processes with supervisor

### PR DESCRIPTION
**Pull request by jleveque**
**- What I did**
 - Ensure all teamd and teamsyncd processes inside docker-teamd are managed by supervisor to ensure we receive syslog messages if any of these processes exit unexpectedly.

**- How I did it**
 - Generate supervisor config file dynamically from a Jinja2 template using docker-teamd-init.sh to accommodate variances in LAG configuration.

**- How to verify it**
 - Start / stop teamd docker (via `systemctl start/stop teamd`) and ensure all processes start/stop properly, and that teamd processes clean up after themselves properly.
- Kill teamd process(es) via signals and ensure supervisor writes appropriate syslog messages.

**- A picture of a cute animal (not mandatory but encouraged)**

```
¸.·´¯`·.´¯`·.¸¸.·´¯`·.¸><(((º>
```

### CODE REVIEW

File: dockers/docker-teamd/docker-teamd-init.sh [hunk 1]
```diff
@@ -25,11 +25,18 @@ for pc in `sonic-cfggen -d -v "PORTCHANNEL.keys() | join(' ') if PORTCHANNEL"`;
...
-
-supervisorctl start rsyslogd
-
-supervisorctl start teamd
+# Create a Python dictionary where the key is the Jinja2 variable name
+# "lags" and the value is a list of dctionaries containing the name of
+# the LAG and the path of the LAG config file. Then output this in
+# JSON format, as we will pass it to sonic-cfggen as additional data
+# below for generating the supervisord config file.
+LAG_INFO_DICT=$(python -c "import json,os,sys; out_dict = {}; out_dict['lags'] = [{'name': os.path.basename(file).split('.')[0], 'file': os.path.join('${TEAMD_CONF_PATH}', file)} for file in os.listdir('${TEAMD_CONF_PATH}')]; sys.stdout.write(json.dumps(out_dict))")
```
Comment (stcheng) [inline dockers/docker-teamd/docker-teamd-init.sh]: Could you also provide a sample output here?

**Comment by stcheng** [comment]
can you also add corresponding j2 tests?

**Comment by jleveque** [comment]
Retest this please.

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Pull request merged**